### ISSUE TITLE
HGCAL: Distance cut missing from 2DCluster's position calculation

### ISSUE THREAD

**Issue by b-fontana** [cms-sw/cmssw#31552]
Upon inspecting [the calculation of 2DClusters positions in release](https://github.com/cms-sw/cmssw/blob/master/RecoLocalCalo/HGCalRecProducers/plugins/HGCalCLUEAlgo.cc#L182-L237), I realized no cut has been so far applied that limits the hits being considered to a radius of 13mm. The cut is however present in the [ImagingAlgo code](https://github.com/cms-sw/cmssw/blob/master/RecoLocalCalo/HGCalRecProducers/plugins/HGCalImagingAlgo.cc#L222-L223). This represents a problem unless the plan is to filter the hits before passing them to the ```HGCalCLUEAlgoT::calculatePosition()``` function.

*Note*: The implemented log-weighted 2DCluster position estimate is explained [here](https://indico.cern.ch/event/776790/contributions/3230583/attachments/1761511/2865143/positionResolutionFor2DClusters_amartell_29Nov.pdf).

**Comment by makortel** [comment]
assign reconstruction

**Comment by perrotta** [comment]
@felicepantaleo @rovere

**Comment by b-fontana** [comment]
Upon further studies with 2018 testbeam data I observed, as expected, that the missing cut does not change the results dramatically. However, using the cut consistently leads to a slightly better spatial resolution. 
Just as an example, I am showing the resolution along X for a positron beam of 50GeV in layer 14 (CE-E) for no distance cut (all hits in each cluster are used)

![inf](https://user-images.githubusercontent.com/20703947/95486844-d7c84180-0993-11eb-865f-540d13390242.png)

and applying a cut at 1.3cm around the most energetic cell of the cluster:

![13](https://user-images.githubusercontent.com/20703947/95486848-d860d800-0993-11eb-9bbf-772f8b404068.png)

The plots above were produced after applying data cleaning as recommended by the testbeam data group and after removing all clusters with less than 5 hits.
The cut was implemented as a [standalone version](https://github.com/b-fontana/TestBeamReconstruction/blob/master/DataProcessing/src/CLUEAnalysis.cc#L72) and should be added to CMSSW.
